plugin_interact.go

```diff
@@ -0,0 +1,57 @@
+package proxy
+
+import (
+	"sync"
+
+	"github.com/anon55555/mt"
+)
+
+// A InteractionHandler holds information on how to handle a Minetest Interaction.
+type InteractionHandler struct {
+	Type        mt.Interaction // can be 255 to register on all interactions
+	Handler     func(*ClientConn, *mt.ToSrvInteract) bool
+}
+
+var interactionHandlers []InteractionHandler
+var interactionHandlerMu sync.RWMutex
+var interactionHandlerOnce sync.Once
+
+// RegisterInteractionHandler adds a new InteractionHandler.
+func RegisterInteractionHandler(handler InteractionHandler) {
+	initInteractionHandlers()
+
+	chatCmdsMu.Lock()
+	defer chatCmdsMu.Unlock()
+
+	interactionHandlers = append(interactionHandlers, handler)
+}
+
+func initInteractionHandlers() {
+	chatCmdsOnce.Do(func() {
+		interactionHandlerMu.Lock()
+		defer interactionHandlerMu.Unlock()
+	
+		interactionHandlers = make([]InteractionHandler, 0)
+	})
+}
+
+func handleInteraction(cmd *mt.ToSrvInteract, cc *ClientConn) bool {
+	handled := false
+	handle := func(cond bool) {
+		if(cond) {
+			handled = true
+		}
+	}
+
+	for _, handler := range interactionHandlers {
+		if handler.Type == 255 {
```

Comment:
I'd prefer a constant named Any(Interaction) as it’s easier to deal with
Constants are defined above global variables (style)
And a custom data type would help with finding the constant in the documentation
If mt already has something like this, use it, otherwise copy or let me do it